### ISSUE TITLE
test code（or pre-trained model weights） is error or not?

### ISSUE THREAD

**Issue by wang-ironman** [Orfium/bytecover#6]
Thank you very much for your excellent work. 

However, when testing the pre-trained model you provided, I found that the MR1: 1270 and mAP: 0.0065 are both very low. There may be some issues somewhere. 

I generated the CSV file myself because some data in SHS100K were not available. So I would like to ask if you have actually tested the testing part of the code and if there are any issues with it. If there are no issues, it may be a problem with my CSV file. 

I would greatly appreciate your response.

**Comment by wang-ironman** [comment]
Hello！

I found that there might be a problem with the pre-trained model after running some tests (such as on the cover80 dataset). 

Could you please check it when you have time? 

Thank you again.

**Comment by progsi** [comment]
Thanks @petvav for providing the pretrained model! Any idea why the mAP is quite low with 0.28 or 0.33 as mentioned in your other [thread](https://github.com/Orfium/bytecover/issues/2)? Regarding your assumptions about implementation details not mentioned in the original paper, I thought maybe (semi)-hard triplet mining instead of sampling might help.

**Comment by wang-ironman** [comment]
> Hello @wang-ironman, 你好 ，
> 
> Thank you for your interest. Can you check if the [debug](https://github.com/Orfium/bytecover/blob/master/config/config.yaml#L5) option is True? If yes, the dataloader generates dummy data for debugging purposes, so you shall set it to False (see [README](https://github.com/Orfium/bytecover#usage)). I have tested the model and the results are:感谢您的关注。你能检查一下 debug 选项是否为 True 吗？如果是，数据加载器会生成用于调试目的的虚拟数据，因此您应将其设置为 False（请参阅自述文件）。我测试了模型，结果是：
> 
> ```
> "mr1": 17.19
> "mAP": 0.28
> ```
> 
> so there is no problem with the pre-trained model.所以预训练的模型没有问题。

Thank you very much for your reply. "debug=False" is set in my code, so I will check my code and training files again.

**Comment by wang-ironman** [comment]
> Thanks @petvav for providing the pretrained model! Any idea why the mAP is quite low with 0.28 or 0.33 as mentioned in your other [thread](https://github.com/Orfium/bytecover/issues/2)? Regarding your assumptions about implementation details not mentioned in the original paper, I thought maybe (semi)-hard triplet mining instead of sampling might help.感谢您提供预训练模型！知道为什么 mAP 相当低，如您在其他线程中提到的 0.28 或 0.33 吗？关于原始论文中未提及的有关实现细节的假设，我认为也许（半）硬三元组挖掘而不是采样可能会有所帮助。

Hello, have you evaluated the provided pre-trained model on the SHS100K test set? If so, what were the results? 

I would greatly appreciate your reply.

**Comment by progsi** [comment]
@wang-ironman I haven't evaluated the model myself yet. The values are provided by the repo owner in the issues. Hoever, I plan to evaluate the model soon myself :)

**Comment by progsi** [comment]
@wang-ironman 

I now evaluated the model on the SHS100K-Test subset that I have (2486 songs). Results are lower than what I reproduce with CQTNet and Re-MOVE for the same subset. 

`Support: 2486
mAP: 0.259
MRR: 0.636`

**Comment by wang-ironman** [comment]
> @wang-ironman
> 
> I now evaluated the model on the SHS100K-Test subset that I have (2486 songs). Results are lower than what I reproduce with CQTNet and Re-MOVE for the same subset.我现在在我拥有的 SHS100K-Test 子集（2486 首歌曲）上评估了模型。结果低于我使用 CQTNet 和 Re-MOVE 对同一子集重现的结果。
> 
> Using `f_c`embedding:使用 `f_c` 嵌入： `mAP: 0.259 MRR: 0.636`
> 
> Using `f_t` embedding:使用 `f_t` 嵌入： `mAP: 0.327 MRR: 0.690 `

Thank you very much for your reply. It's very helpful to me.

**Comment by progsi** [comment]
@wang-ironman you are welcome!
@petvav I just discovered this paper. The authors also tried to implement ByteCover and discovered a range of training tricks which helped to improve the performance as a baseline. Additionally, the authors omit the GeM module: https://arxiv.org/abs/2306.09025
